%\VignetteIndexEntry{An overview of yeastRNASeq}
%\VignetteDepends{yeastRNASeq,ShortRead}
%\VignettePackage{yeastRNASeq}

\documentclass[12pt]{article}

%%%%%%%%%%%%%%%%%%%%%%%% Standard Packages %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\usepackage{times}
\usepackage{hyperref}

\newcommand{\RPackage}[1]{\textsf{#1}}
\newcommand{\RCode}[1]{\texttt{#1}}
\newcommand{\RObject}[1]{\texttt{#1}}
\newcommand{\RFunction}[1]{\textsf{#1}}
\newcommand{\RClass}[1]{{\textsf{#1}}}
\SweaveOpts{eps=FALSE, echo=TRUE}

\title{An overview of yeastRNASeq}
\author{James Bullard and Kasper D.\ Hansen}
\date{Modified: 15 January 2010. Compiled: \today}

\begin{document}
\maketitle

<<echo=FALSE>>=
options(width = 65)
@ 

<<>>=
require(yeastRNASeq)
@ 

This package contains data from
<<>>=
citation(package = "yeastRNASeq")
@ 
which describes some experiments in \emph{S.\ cerevisiae} comparing
various mutant strains to a wild-type strain.  This package contains
data from a wild-type and a single mutant.  For each condition
(mutant, wild-type) there is two lanes worth of data, each lane
containing a sample of 500,000 raw (unaligned) reads from each of 2
lanes each.  Each of the four lanes have been aligned against the
yeast genome using Bowtie. 

The full set of raw FASTQ and Bowtie alignment files are
<<>>=
list.files(file.path(system.file(package = "yeastRNASeq"), "reads"))
@ 
These files are ready to be parsed by the tools in the \RPackage{ShortRead}
package.  As an example we read the alignment files by
<<>>=
require(ShortRead)
files <- list.files(file.path(system.file(package = "yeastRNASeq"), "reads"),
                    pattern = "bowtie", full.names = TRUE)
names(files) <- gsub("\\.bowtie.*", "", basename(files))
aligned <- lapply(files, readAligned, type = "Bowtie")
@ 
The constructed object is a list with 4 elements.  Each element
correspond to a lane and is an object of class \RClass{AlignedRead}.

The output from this operation has already been stored as an R object
and is accessible by
<<>>=
data(yeastAligned)
@ 
The alignment percent is
<<>>=
sapply(aligned, length) / 500000
@ 

There are two additional objects available in the package, purely for
illustrative purposes (do not use them for analysis).  The object
\RObject{yeastAnno} is annotation obtained from Ensembl using biomaRt
and is a \RClass{data.frame} of annotation:
<<>>=
data(yeastAnno)
head(yeastAnno)
table(yeastAnno$gene_biotype)
@ 
The other object is called \RObject{geneLevelData} and is a
\RClass{matrix} of counts per gene.
<<>>=
data(geneLevelData)
head(geneLevelData)
@ 

\section*{Some details}

The FASTQ files in the package are using the old-style (by now,
probably deprecated) numeric qualities, instead of ASCII based qualities:
<<>>=
fastq <- list.files(file.path(system.file(package = "yeastRNASeq"), "reads"),
                    pattern = "wt_1_f.fastq")
@ 

The reads were aligned to the yeast genome obtained from
\url{ftp://ftpmips.gsf.de/yeast/sequences} (which was the basis for
the Bowtie index available at the Bowtie website at the time of
alignment).


\section*{Old}

There are some files available. 
<<eval=FALSE>>=
require(yeastRNASeq)
files <- list.files(file.path(system.file(package = "yeastRNASeq"), "reads"), full.names = TRUE, pattern = "bowtie$")
files
@ 

Some have been aligned with bowtie. You can read these in using
\RPackage{ShortRead}. 

<<echo=TRUE,eval=FALSE>>=
require(ShortRead)
aligned <- lapply(files, function(f) readAligned(f, type = "Bowtie"))
names(aligned) <- gsub(".fastq.bowtie", "", basename(files))
@ 

<<echo=TRUE,eval=FALSE>>=
sapply(aligned, function(a) table(chromosome(a)))
@ 

<<fig=TRUE, echo=TRUE,eval=FALSE>>=
a <- alphabetByCycle(sread(aligned[[1]]))
matplot(t(a), ylab = "reads", xlab = "cycle", type = "l")
@ 



\end{document}
